kernel/sorts.ml

```diff
@@ -25,20 +25,25 @@ let prop = Prop
 let set = Set
 let type1 = Type type1_univ
 
-let univ_of_sort = function
-  | Type u -> u
-  | Set -> Universe.type0
-  | Prop -> Universe.type0m
-  | SProp -> Universe.sprop
-
 let sort_of_univ u =
   if Universe.is_sprop u then sprop
   else if is_type0m_univ u then prop
   else if is_type0_univ u then set
   else Type u
 
 let compare s1 s2 =
-  if s1 == s2 then 0 else Universe.compare (univ_of_sort s1) (univ_of_sort s2)
+  if s1 == s2 then 0 else
+    match s1, s2 with
+    | SProp, SProp -> 0
+    | SProp, _ -> -1
+    | _, SProp -> 1
+    | Prop, Prop -> 0
+    | Prop, _ -> -1
+    | Set, Prop -> 1
+    | Set, Set -> 0
+    | Set, _ -> -1
+    | Type u1, Type u2 -> Universe.compare u1 u2
+    | Type _, _ -> -1
```

Comment:
Am I reading this right that Type,Set and Set,Type both produce -1?

Reply:
This change is a revert from the previous PR. I've realized in the meantime that it is guaranteed that Type@{Set} never happens thanks to the private type and the smart constructors, so this is actually fine.

Reply:
Ah, but you're right there is something wrong even with the Type@{Set} trick. Will fix.